zerver/openapi/python_examples.py

```diff
@@ -216,6 +216,79 @@ def get_members(client):
         else:
             assert member.get('profile_data', None) is not None
 
+@openapi_test_function("/users/{user_id}:get")
+def get_single_user(client):
+    # type: (Client) -> None
+
+    # {code_example|start}
+    # Get user data of a user
+    user_id = 8
+    url = 'users/' + str(user_id)
+    result = client.call_endpoint(
+        url=url,
+        method='GET'
+    )
+    # {code_example|end}
+    validate_against_openapi_schema(result, '/users/{user_id}', 'get', '200')
+
+    # {code_example|start}
+    # You may pass the `client_gravatar` query parameter as follows:
+    result = client.call_endpoint(
+        url=url,
+        method='GET',
+        request={'client_gravatar': True}
+    )
+    # {code_example|end}
+
+    validate_against_openapi_schema(result, '/users/{user_id}', 'get', '200')
+    assert result['members'][0]['avatar_url'] is None
+
+    # {code_example|start}
+    # You may pass the `include_custom_profile_fields` query parameter as follows:
+    result = client.call_endpoint(
+        url=url,
+        method='GET',
+        request={'include_custom_profile_fields': True}
+    )
+    # {code_example|end}
+
+    validate_against_openapi_schema(result, '/users/{user_id}', 'get', '200')
+
+@openapi_test_function("/users/{user_id}:delete")
+def deactivate_user(client):
+    # type: (Client) -> None
+
+    # {code_example|start}
+    # Get user data of a user
+    user_id = 8
+    url = 'users/' + str(user_id)
+    result = client.call_endpoint(
+        url=url,
+        method='DELETE',
+    )
```

Comment:
This isn't necessarily a blocker for merging this, but we should definitely do a PR in the `python-zulip-api` adding native functions for these queries so we don't have to use `call_endpoint`; `call_endpoint` is meant to be the workaround for endpoints that aren't properly setup.  It might be less work to merge that first so we only need to get the final version for these examples right; do you want to try doing that?

Reply:
@timabbott Okay, I'll do that as soon as I complete work on my pending PRs.